src/main/java/org/apache/daffodil/DaffodilSaver.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil;
+
+import java.lang.reflect.Method;
+import java.nio.channels.FileChannel;
+import java.nio.channels.WritableByteChannel;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.net.URI;
+import java.net.URL;
+import java.util.List;
+
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.DocumentBuilder;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+import org.w3c.dom.Node;
+
+// We need a special customized classpath, and the easiest way to do that within SBT is by
+// forking. But the only thing that can save a parser via forking is the Daffodil CLI, which we
+// do not publish. So instead we create this class that has a static main that we can fork to
+// compile and save schemas using the version of Daffodil that is on the classpath. Note that it
+// also uses reflection so that it is not tied to any specific version of Daffodil. This is
+// fragile, but the Daffodil Jave API is pretty set in stone at this point, so this reflection
```

Comment:
Jave eh?

Reply:
It will help avoiding similar issues when converting to Scala 3, though I guess there is some compatibility between 2.13 and 3 as far as libraries and dependencies go I think.

Reply:
Yeah, it's a pain for sure. But SBT plugins are always compiled with 2.12, and there isn't a way I could find to have SBT fork DaffodilSaver but bootstrap with a specific version of scala, it just uses whatever `scalaVersion` is set to. So we either need to ensure DaffodilSaver.scala is always compiled in a way that is compatible with all versions of Scala (which is fragile, if its even possible) or just drop the scala requirement altogether.

Reply:
> It will help avoiding similar issues when converting to Scala 3, though I guess there is some compatibility between 2.13 and 3 as far as libraries and dependencies go I think.

Just FYI: There's not a lot of source code compatibility in Scala 2.13 to 3 even with the compatibility feature for scala 3 enabled. I did some experimentation: https://issues.apache.org/jira/browse/DAFFODIL-2975